crates/astria-conductor/src/sequencer/mod.rs

```diff
@@ -220,29 +216,37 @@ impl RunningReader {
                     // otherwise recover from a failed block fetch.
                     let block = block.wrap_err("the stream of new blocks returned a catastrophic error")?;
                     if let Err(error) = self.block_cache.insert(block) {
-                        warn!(%error, "failed pushing block into sequential cache, dropping it");
+                        warn_span!("conductor::sequencer::RunningReader::run_loop").in_scope(||
+                            warn!(%error, "failed pushing block into sequential cache, dropping it")
+                        );
                     }
                 }
 
                 // Record the latest height of the Sequencer network, allowing `blocks_from_heights` to progress.
                 Some(res) = self.latest_height_stream.next() => {
-                    match res {
-                        Ok(height) => {
-                            debug!(%height, "received latest height from sequencer");
-                            self.blocks_from_heights.set_latest_observed_height_if_greater(height);
-                        }
-                        Err(error) => {
-                            warn!(
-                                error = %Report::new(error),
-                                "failed fetching latest height from sequencer; waiting until next tick",
-                            );
-                        }
-                    }
+                    self.latest_height_stream_handler(res);
                 }
             }
         }
     }
 
+    #[instrument(skip_all)]
+    fn latest_height_stream_handler(&mut self, res: Result<Height, tendermint_rpc::Error>) {
```

Comment:
Also could be named just `handle_latest_height`?

Reply:
Renamed in 47b2a8d. Thanks for taking a look!